### PR TITLE
updated tests with new transaction builder approach

### PR DESCRIPTION
**Pull request by nothingalike**
@tweakch  here is my approach for Transaction Building, let me know what you think.

I updated all tests with the new approach so you can see it in action.

### CODE REVIEW

File: CardanoSharp.Wallet/TransactionBuilding/ABuilder.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CardanoSharp.Wallet.TransactionBuilding
+{
+    public abstract class ABuilder<T>
```
Comment (tweakch) [inline CardanoSharp.Wallet/TransactionBuilding/ABuilder.cs:7]: Maybe a `ACollectionBuilder<T>` that provides a `BuildAll()` method could reduce the `Build`-Tails
Or a set of `ACompositeBuilder<T1,T2,T3,...>` with two, three, four, five generic types...

**Review by tweakch — APPROVED** [review]
Well done. Do you think this could be useful for cscli?

File: CardanoSharp.Wallet.Test/TransactionTests.cs [hunk 2]
```diff
@@ -57,40 +58,34 @@ public void SerializeTransaction()
...
+                            .WithTransactionOutputValue(new TransactionOutputValueBuilder()
+                                .WithCoin(amount)
+                                .Build())
+                            .Build(),
+                        new TransactionOutputBuilder()
+                            .WithAddress(payment2Addr.GetBytes())
+                            .WithTransactionOutputValue(new TransactionOutputValueBuilder()
+                                .WithCoin(amount)
+                                .Build())
+                            .Build()
```
[the remark this answers is not in the document]
Comment (nothingalike) [reply CardanoSharp.Wallet.Test/TransactionTests.cs:84]: Yeah collections are odd. honestly what i dont like the most are the instances where its not really a type, any of the `Dictionary<>`. They probably could be turned into types, but can be done later

File: CardanoSharp.Wallet.Test/TransactionTests.cs [hunk 3]
```diff
@@ -170,40 +165,32 @@ public void BuildTxWithChange()
...
-                            Coin = 856488
-                        }
-                    }
-                },
-                Ttl = 1000,
-                Fee = 143502
-            };
+                    new TransactionOutputBuilder()
+                        .WithAddress(baseAddr.GetBytes())
+                        .WithTransactionOutputValue(new TransactionOutputValueBuilder()
```
[the remark this answers is not in the document]
Comment (nothingalike) [reply CardanoSharp.Wallet.Test/TransactionTests.cs:180]: yeah large/complicated objects definitely have that affect

File: CardanoSharp.Wallet/TransactionBuilding/ABuilder.cs (hunk 1 above)
Comment (nothingalike) [reply CardanoSharp.Wallet/TransactionBuilding/ABuilder.cs:7]: yeah we can play around with different mechanics to help clean up the syntax

**Pull request merged**